ambry-admin/src/main/java/com.github.ambry.admin/AdminBlobStorageService.java

```diff
@@ -307,47 +292,50 @@ private void checkAvailable()
   private class InboundIdConverterCallback implements Callback<String> {
     private final RestRequest restRequest;
     private final RestResponseChannel restResponseChannel;
-    private final HeadForGetCallback headForGetCallback;
-    private final HeadCallback headCallback;
-    private final DeleteCallback deleteCallback;
     private final CallbackTracker callbackTracker;
 
+    private HeadForGetCallback headForGetCallback = null;
+    private HeadCallback headCallback = null;
+    private DeleteCallback deleteCallback = null;
+
     InboundIdConverterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel,
         HeadForGetCallback callback) {
-      this(restRequest, restResponseChannel, callback, null, null);
+      this(restRequest, restResponseChannel);
+      headForGetCallback = callback;
     }
 
     InboundIdConverterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel,
         HeadCallback callback) {
-      this(restRequest, restResponseChannel, null, callback, null);
+      this(restRequest, restResponseChannel);
+      headCallback = callback;
     }
 
     InboundIdConverterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel,
         DeleteCallback callback) {
-      this(restRequest, restResponseChannel, null, null, callback);
+      this(restRequest, restResponseChannel);
+      deleteCallback = callback;
     }
 
-    private InboundIdConverterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel,
-        HeadForGetCallback headForGetCallback, HeadCallback headCallback, DeleteCallback deleteCallback) {
+    InboundIdConverterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel) {
       this.restRequest = restRequest;
       this.restResponseChannel = restResponseChannel;
-      this.headForGetCallback = headForGetCallback;
-      this.headCallback = headCallback;
-      this.deleteCallback = deleteCallback;
       callbackTracker = new CallbackTracker(restRequest, OPERATION_TYPE_INBOUND_ID_CONVERSION,
           adminMetrics.inboundIdConversionTimeInMs, adminMetrics.inboundIdConversionCallbackProcessingTimeInMs);
       callbackTracker.markOperationStart();
     }
 
     /**
      * Forwards request to the {@link Router} once ID conversion is complete.
+     * </p>
+     * In the case of some sub-resources, executes the operation without forwarding to the {@link Router}.
```

Comment:
"In the case of some sub-resources (e.g., replicas), the request is not forwarded to the {@link Router}."